bpf/bpf_lxc.c

```diff
@@ -1931,6 +2002,28 @@ ipv4_policy(struct __ctx_buff *ctx, struct iphdr *ip4, __u32 src_label,
 		if (unlikely(ct_state->rev_nat_index)) {
 			int ret2;
 
+#ifdef ENABLE_DSR
+			if (ct_state->dsr_internal) {
+				struct ipv4_ct_tuple dsr_tuple = {};
+				struct ipv4_nat_entry *dsr_entry;
+
+				dsr_tuple = *tuple;
+				dsr_tuple.flags = NAT_DIR_EGRESS;
+				dsr_tuple.sport = tuple->dport;
+				dsr_tuple.dport = tuple->sport;
+
+				dsr_entry = nodeport_dsr_lookup_v4_nat_entry(&dsr_tuple);
+				if (dsr_entry) {
+					ret = __lb4_rev_nat(ctx, ETH_HLEN, l4_off, tuple,
+							    &dsr_entry->nat_info,
+							    ct_state->loopback,
+							    ipfrag_has_l4_header(fraginfo));
+					if (IS_ERR(ret))
+						return ret;
+					break;
+				}
```

Comment:
I think we need to do better here, and not have two near-identical calls to `lb4_rev_nat()`.